plugins/jobs/src/js/components/JobFormModal.tsx

```diff
@@ -102,39 +145,251 @@ export default class JobFormModal extends React.Component<
     isEdit: false,
     job: new Job()
   };
+  private triggerFormSubmit: FormSubmitCallback;
 
   constructor() {
-    super(...arguments);
+    super();
 
     this.state = {
-      defaultTab: ""
+      jobFormModel: null,
+      jobJsonString: null,
+      isJsonMode: false
     };
 
-    this.handleTabChange = this.handleTabChange.bind(this);
+    this.triggerFormSubmit = undefined;
```

Comment:
🎨 I'd prefer `null` here. 

If I get it correct, this method is _not yet_ set, but there is no legit case where it can be set to `undefined` _once it is set_. In these cases, I'd use `null`. 
`undefined` IMHO is better for params that can be skipped or something that also can be (re)set to  "nothing" during code execution. 🤔 

That said, this usage of `null` comes from languages which only have `null`, but no additional `undefined` 😅Also it is more a gut feeling without me knowing if there is any convention/best practice in Javascript